### PR TITLE
Make diff report only files changed on this branch

### PR DESCRIPTION
**Pull request by bklang**
I noticed this behavior when working with a very small pull request, but which took a long time to review. For purposes of discussion, imagine the following timeline:

* Monday, create a branch off of `main` and make a documentation change (eg. update `README.md`)
* Tuesday, Wednesday, Thursday: other teams make commits unrelated to mine, which get merged into the `main` branch
* In an effort to keep my branch up-to-date, I merge `main` into my branch

After this sequence of events, cobra commander now reports that my branch includes not only my change to `README.md`, but also *all of the changes* from all of the other things merged into `main` since I originally created my branch. This feels like it is overreporting the scope of my PR, and will cause CI to test many additional components that need not be tested.

This change proposes modifying the git check to only look for files which are different on this PR vs. the base branch.

### CODE REVIEW

**Comment by benlangfeld** [comment]
> This change proposes modifying the git check to only look for files which are different on this PR vs. the base branch.

File: lib/cobra_commander/git_changed.rb [hunk 1]
```diff
@@ -27,7 +27,7 @@ def each(&block)
     def changes
       @changes ||= begin
         diff, _, result = Dir.chdir(@repo_root) do
-          Open3.capture3("git", "diff", "--name-only", @base_branch)
+          Open3.capture3("git", "diff", "--name-only", "--merge-base", @base_branch)
```
Comment (benlangfeld) [inline lib/cobra_commander/git_changed.rb:30]: I think we need some test coverage to demonstrate the bug.

Comment (bklang) [reply lib/cobra_commander/git_changed.rb:30]: Fixture modified to demonstrate the bug

**Review by xjunior — APPROVED** [review]

**Comment by xjunior** [comment]
@bklang your spec commit isn't signed. I'll rebase this PR on `main` to sign all commits.

**Review by xjunior — APPROVED** [review]

**Comment by benlangfeld** [comment]
nooooo. This is wrong. We should revert it.

**Pull request merged**